### DISCUSSION TITLE
solution type

### DISCUSSION THREAD

**Discussion by dingqiushi** [idaholab/moose#22412] [Q&A General] [answered]
I would like to know whether the expression of the solution is of nodal or modal type. With my current knowledge of the code, I think the type of the variable in the configuration file is nodal when set to lagrange and modal when set to monomial. I don't know if I understand this correctly. Is there a specific document that says something about nodal 
representation and modal representation?

**Comment by GiudGiud** [comment, marked as answer]
Hello

There's a lot of finite element families supported and whether a variable is 'nodal' or 'elemental' isnt always the best description as a variable can have degrees of freedom on both nodes and elements (especially as libmesh often stores shared dofs whether nodal, edge or face-based dofs, on the nodes). 
I dont think we have a documentation page on all the families supported. There's an enum there:
https://libmesh.github.io/doxygen/namespacelibMesh.html#abcdbcd8364bf9fb7a67b3b4013031b2d
The nodal ones are lagrange and hierarchic iirc. 

Within the code, you can perform a call to var->isNodal() (from a MooseVariable* pointer) to see if a variable is nodal 

Guillaume
